griptape/tools/openweather_client/tool.py

```diff
@@ -0,0 +1,147 @@
+        }
+    )
+    def get_coordinates_by_location(self, params: dict) -> ListArtifact | TextArtifact | ErrorArtifact:
+        location = params["values"].get("location")
+        coordinates = self._fetch_coordinates(location)
+        if coordinates:
+            lat, lon = coordinates
+            return TextArtifact(f"Coordinates for {location}: Latitude: {lat}, Longitude: {lon}")
+        else:
+            return ErrorArtifact(f"Error fetching coordinates for location: {location}")
+
+    def _fetch_coordinates(self, location: str) -> tuple[float, float] | None:
+        US_STATE_CODES = ["AL", "AK", "AZ", "AR", "CA", "CO", "CT", "DE", "FL", "GA", "HI", "ID", "IL", "IN", "IA", "KS", "KY", "LA", "ME", "MD", "MA", "MI", "MN", "MS", "MO", "MT", "NE", "NV", "NH", "NJ", "NM", "NY", "NC", "ND", "OH", "OK", "OR", "PA", "RI", "SC", "SD", "TN", "TX", "UT", "VT", "VA", "WA", "WV", "WI", "WY"]
+        parts = location.split(',')
+        if len(parts) == 2 and parts[1].strip() in US_STATE_CODES:
+            location += ', US'
+        request_params = {
+            'q': location,
+            'limit': 1,
+            'appid': self.api_key
+        }
+        try:
+            response = requests.get(self.GEOCODING_URL, params=request_params)
+            if response.status_code == 200:
+                data = response.json()
+                if data and isinstance(data, list):
+                    return data[0]['lat'], data[0]['lon']
+            else:
+                logging.error(f"Error fetching coordinates. HTTP Status Code: {response.status_code}. Response: {response.text}")
+        except Exception as e:
+            logging.error(f"Error fetching coordinates: {e}")
+        return None
+
+
+    @activity(
+        config={
+            "description": "Can be used to fetch current weather data for a given location using the OpenWeather API. Temperatures are returned in {{ units }} by default.",
+            "schema": Schema({
+                Literal("location", description="Location to fetch weather data for."): str
+            }),
+        }
+    )
+    def get_current_weather_by_location(self, params: dict) -> ListArtifact | TextArtifact | ErrorArtifact:
+        location = params["values"].get("location")
+        coordinates = self._fetch_coordinates(location)
+        if coordinates:
+            lat, lon = coordinates
+            request_params = {
+                'lat': lat,
+                'lon': lon,
+                'exclude': 'minutely,hourly,daily,alerts',
+                'appid': self.api_key,
+                'units': self.units
+            }
+            return self._fetch_weather_data(request_params)
+        else:
+            return ErrorArtifact(f"Error fetching coordinates for location: {location}")
+
+    @activity(
+        config={
+            "description": "Fetch hourly forecast for a given location up to 48 hours ahead using the OpenWeather API. Temperatures are returned in {{ units }} by default.",
```

Comment:
1. Should be "Can be used..."
2. Drop "OpenWeather API"
3. `{{ _self.units }}`